dashboard/app/models/plc/user_course_enrollment.rb

```diff
@@ -74,6 +75,25 @@ def create_enrollment_unit_assignments
     end
   end
 
+  def create_authorized_teacher_user_permission
+    unless user.authorized_teacher?
+      UserPermission.create(
+        user_id: user_id,
+        permission: UserPermission::AUTHORIZED_TEACHER
+      )
+    end
+  end
+
+  def delete_authorized_teacher_user_permission
+    # De-authorize teacher unless they have other plc enrollments
+    unless user.plc_enrollments.length > 1
+      UserPermission.where(
+        user_id: user_id,
+        permission: UserPermission::AUTHORIZED_TEACHER
+      ).destroy_all
+    end
+  end
+
```

Comment:
It would be great see tests in `UserCourseEnrollmentTest` verifying these methods (callbacks) work as expected.